### PR TITLE
add usage example for hooks

### PR DESCRIPTION
**Pull request by chinwobble**
The type signatures for the hooks are quite hard to read.

Especially since most parts of it are abstracted away by the brisk-reconciler.
This change makes it easier to discover how to use the common hooks.

image

### CODE REVIEW

File: src/UI_Hooks/Reducer.rei [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+{2 Usage:}
+
+{[
+let%hook (state, dispatch) = Hooks.reducer(~initialState={counter:0}, reduce);
+]}
+*/
 let reducer:
   (~initialState: 'a, ('b, 'a) => 'a, t(Reducer.t('a) => 'c, 'd)) =>
-  (('a, 'b => unit), t('c, 'd));
+  (('a, 'b => unit), t('c, 'd));
```
Comment (Et7f3) [inline src/UI_Hooks/Reducer.rei:16]: Can you run `esy format`

File: src/UI_Hooks/Reducer.rei [hunk 2]
```diff
@@ -1,5 +1,16 @@
 open Revery_UI.React.Hooks;
 
+/**
+{2 Description:}
+
+TODO
```
Comment (Et7f3) [inline src/UI_Hooks/Reducer.rei:6]: Description is mostly the same than react.js

File: src/UI_Hooks/Effect.rei [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
 
+/**
+{2 Description:}
+
+The Effect Hook lets you perform side effects in function components.
+
+{2 Usage:}
+
+{[
+let%hook (state, dispatch) = Hooks.effect(OnMount, reduce);
```
Comment (lessp) [inline src/UI_Hooks/Effect.rei:11]:
```suggestion
let%hook () = Hooks.effect(OnMount, () => {
  // Some side-effect

  // Clean-up function
  None;
});
```

Comment (chinwobble) [reply src/UI_Hooks/Effect.rei]: Great thanks. I didn't know about the suggested change feature on github.

**Review by chinwobble — COMMENTED** [review]
Implemented the PR feedback.

File: src/UI_Hooks/Reducer.rei [hunk 4]
```diff
@@ -3,7 +3,7 @@ open Revery_UI.React.Hooks;
 /**
 {2 Description:}
 
-TODO
+An alternative to Hooks.state. Accepts a reducer of type (state, action) => newState, and returns the current state paired with a dispatch function.
```
Comment (Et7f3) [inline src/UI_Hooks/Reducer.rei:6]:
```suggestion
An alternative to Hooks.state. Accepts a reducer of type (action, state) => newState, and returns the current state paired with a dispatch function.
```
the type is: `('b, 'a) => 'a`

**Review by Et7f3 — APPROVED** [review]
Wait for CI to merge.

**Comment by bryphe** [comment]
Thanks @chinwobble for the PR, and thanks @Et7f3 for reviewing! 👍

**Pull request merged**